bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java

```diff
@@ -1162,6 +1165,59 @@ private CaseDbSchemaVersionNumber updateFromSchema7toSchema7dot1(CaseDbSchemaVer
 		} finally {
 			closeResultSet(resultSet);
 			closeStatement(statement);
+
+		}
+	}
+
+	/**
+	 * Updates a schema version 7.1 database to a schema version 7.2 database.
+	 *
+	 * @param schemaVersion The current schema version of the database.
+	 * @param connection    A connection to the case database.
+	 *
+	 * @return The new database schema version.
+	 *
+	 * @throws SQLException     If there is an error completing a database
+	 *                          operation.
+	 * @throws TskCoreException If there is an error completing a database
+	 *                          operation via another SleuthkitCase method.
+	 */
+	private CaseDbSchemaVersionNumber updateFromSchema7dot1toSchema7dot2(CaseDbSchemaVersionNumber schemaVersion, CaseDbConnection connection) throws SQLException, TskCoreException {
+		if (schemaVersion.getMajor() != 7) {
+			return schemaVersion;
+		}
+
+		if (schemaVersion.getMinor() != 1) {
+			return schemaVersion;
+		}
+
+		Statement statement = null;
+		Statement updstatement = null;
+		ResultSet resultSet = null;
+		try {
+			statement = connection.createStatement();
+			updstatement = connection.createStatement();
+
+			//add the data_source_obj_id column to blackboard_artifacts.
+			statement.execute("ALTER TABLE blackboard_artifacts ADD COLUMN data_source_obj_id INTEGER NOT NULL DEFAULT -1");
+
+			// populate data_source_obj_id for each artifact
+			resultSet = connection.executeQuery(statement, "SELECT artifact_id, obj_id FROM blackboard_artifacts"); //NON-NLS
+			while (resultSet.next()) {
+				long artifact_id = resultSet.getLong("artifact_id");
+				long obj_id = resultSet.getLong("obj_id");
+				long data_source_obj_id = getDataSourceObjectId(connection, obj_id);
+
+				updstatement.executeUpdate("UPDATE blackboard_artifacts SET data_source_obj_id = " + data_source_obj_id + " "
+						+ "WHERE artifact_id = " + artifact_id);
+			}
```

Comment:
We need to ask @bcarrier if he wants to do the data upgrade as well as the schema upgrade. I am generally in favor of always doing a data upgrade to eliminate the possibility of  issues going forward,  and we have a precedent for doing the data upgrade when we did the same denormalization to the tsk_files table (see updateFromSchema3toSchema4), but Brian did not want to do a data upgrade in updateFromSchema6toSchema7, so we should let him make the call.

Reply:
I am commenting out the call to this code so we can proceed. We will resolve this question later.